dali/kernels/common/scatter_gather.h

```
@@ -71,7 +71,7 @@ class DLL_PUBLIC ScatterGatherGPU {
    * @brief Adds one copy to the batch
    */
   void AddCopy(void *dst, const void *src, size_t size) {
-    if (size > 0) {
+    if (size > 0 && dst != nullptr && src != nullptr) {
```

I don't like it being here... It will hide errors.